process/logsevents/logsAndEventsProcessor.go

```diff
@@ -51,74 +60,62 @@ func (lep *logsAndEventsProcessor) ExtractDataFromLogsAndPutInAltered(
 	logsAndEvents map[string]nodeData.LogHandler,
 	preparedResults *data.PreparedResults,
 	timestamp uint64,
-) (data.TokensHandler, tags.CountTags) {
-	txsMap := converters.ConvertTxsSliceIntoMap(preparedResults.Transactions)
-	scrsMap := converters.ConvertScrsSliceIntoMap(preparedResults.ScResults)
+) (data.TokensHandler, tags.CountTags, map[string]*data.ScDeployInfo) {
+	lep.txsMap = converters.ConvertTxsSliceIntoMap(preparedResults.Transactions)
+	lep.scrsMap = converters.ConvertScrsSliceIntoMap(preparedResults.ScResults)
+	lep.tagsCount = tags.NewTagsCount()
+	lep.tokens = data.NewTokensInfo()
+	lep.accounts = preparedResults.AlteredAccts
+	lep.timestamp = timestamp
+	lep.scDeploys = make(map[string]*data.ScDeployInfo)
 
-	tagsCount := tags.NewTagsCount()
-	tokens := data.NewTokensInfo()
 	for logHash, log := range logsAndEvents {
 		if check.IfNil(log) {
 			continue
 		}
 
 		events := log.GetLogEvents()
-		lep.processEvents(logHash, timestamp, events, tokens, tagsCount, preparedResults.AlteredAccts, txsMap, scrsMap)
+		lep.processEvents(logHash, events)
 	}
 
-	return tokens, tagsCount
+	return lep.tokens, lep.tagsCount, lep.scDeploys
 }
 
-func (lep *logsAndEventsProcessor) processEvents(
-	logHash string,
-	timestamp uint64,
-	events []nodeData.EventHandler,
-	tokens data.TokensHandler,
-	tagsCount tags.CountTags,
-	accounts data.AlteredAccountsHandler,
-	txsMap map[string]*data.Transaction,
-	scrsMap map[string]*data.ScResult,
-) {
+func (lep *logsAndEventsProcessor) processEvents(logHash string, events []nodeData.EventHandler) {
 	for _, event := range events {
 		if check.IfNil(event) {
 			continue
 		}
 
-		lep.processEvent(logHash, timestamp, event, tokens, tagsCount, accounts, txsMap, scrsMap)
+		lep.processEvent(logHash, event)
 	}
 }
 
-func (lep *logsAndEventsProcessor) processEvent(
-	logHash string,
-	timestamp uint64,
-	events nodeData.EventHandler,
-	tokens data.TokensHandler,
-	tagsCount tags.CountTags,
-	accounts data.AlteredAccountsHandler,
-	txsMap map[string]*data.Transaction,
-	scrsMap map[string]*data.ScResult,
-) {
+func (lep *logsAndEventsProcessor) processEvent(logHash string, events nodeData.EventHandler) {
 	logHashHexEncoded := hex.EncodeToString([]byte(logHash))
 	for _, proc := range lep.eventsProcessors {
 		identifier, processed := proc.processEvent(&argsProcessEvent{
-			event:     events,
-			accounts:  accounts,
-			tokens:    tokens,
-			tagsCount: tagsCount,
-			timestamp: timestamp,
+			event:            events,
+			accounts:         lep.accounts,
+			tokens:           lep.tokens,
+			tagsCount:        lep.tagsCount,
+			timestamp:        lep.timestamp,
+			scDeploys:        lep.scDeploys,
+			txHashHexEncoded: logHashHexEncoded,
 		})
-		if identifier == "" {
-			continue
+		emptyIdentifier := identifier == ""
+		if emptyIdentifier && processed {
+			return
```

Comment:
is it ok to return here? The old code had a `continue`

Reply:
Yes. In this case is ok to return because `log is processed`.